src/components/TableX.vue

```diff
@@ -1,48 +1,133 @@
-<script lang="ts">
-/*Declearations had to be separated because of volar error*/
-
-type TableCol = {
-    label: string,
-    name: string,
-    className?: string,
-    thClass?: string,
-    width?: string
-};
-interface TableXProps{
-    cols: TableCol[],
-    data: any[]
-}
-</script>
-
-<script setup lang="ts">
-const props = defineProps<TableXProps>();
-/*
-props.cols.forEach((col, idx)=>{
-    if(typeof col === 'string'){
-        props.cols[idx] = {
-            label: col
-        }
-    }
-})
-*/
-</script>
-
-<template>
-    <div>
-        <table>
-            <colgroup>
-                <col v-for="cprops in cols" :width="cprops.width"/>
-            </colgroup>
-            <thead>
-                <tr>
-                    <th v-for="cprops in cols" :class="cprops.thClass">{{ cprops.label }}</th>
-                </tr>
-            </thead>
-            <tbody>
-                <tr v-for="row in data">
-                    <td v-for="cprops in cols" :class="cprops.className">{{ row[cprops.name] }}</td>
-                </tr>
-            </tbody>
-        </table>
-    </div>
-</template>
\ No newline at end of file
+<script setup lang="ts">
+import { onMounted, onUnmounted, onUpdated, ref } from "vue";
+
+export type TableCol = {
+    label: string;
+    name: string;
+    className?: string;
+    thClass?: string;
+    width?: string;
+};
+export interface TableXProps {
+    cols: TableCol[];
+    data: any[];
+}
+
+const tableRef = ref<HTMLTableElement>();
+
+const props = withDefaults<TableXProps, {}>(defineProps<TableXProps>(), {
+    width: "auto",
```

Comment:
This line has an issue. I suppose `width: 'auto'` is meant as the default value for objects in `col` props but this does not achieve that. In fact, I don't see this being used anywhere in the code

Reply:
Sorry about that. Most likely it was for allowing the parent component to set table width. Since we have now forwarded all attributes to table. This is not necessary anymore.

Reply:
It has been there since the pull request was created. Anyway. Remove all unnecessary stuff and push a fix/refactor commit.